stock_move_picking_type_origin/models/stock_rule.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+
+    _inherit = "stock.rule"
+
+    def _get_stock_move_values(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_dest_id,
+        name,
+        origin,
+        company_id,
+        values,
+    ):
+        stock_move_values = super()._get_stock_move_values(
+            product_id,
+            product_qty,
+            product_uom,
+            location_dest_id,
+            name,
+            origin,
+            company_id,
+            values,
+        )
+        move_dest = values.get("move_dest_ids")
+        if move_dest:
+            original_picking_type_id = move_dest[0].original_picking_type_id.id
+        else:
+            original_picking_type_id = stock_move_values.get("picking_type_id")
+        stock_move_values["original_picking_type_id"] = original_picking_type_id
+        return stock_move_values
+
+    def _push_prepare_move_copy_values(self, move_to_copy, new_date):
+        values = super()._push_prepare_move_copy_values(move_to_copy, new_date)
+        values["original_picking_type_id"] = move_to_copy.original_picking_type_id.id
```

Comment:
@sbejaoui 
????
```suggestion
        values["original_picking_type_id"] = move_to_copy.original_picking_type_id.id or move_to_copy.picking_type_id.id
```
?????